module-check/src/main/clojure/clojure/core/typed/base_env.clj

```diff
@@ -740,6 +740,8 @@ clojure.string/trim [String -> String]
 clojure.string/trimr [String -> String]
 clojure.string/triml [String -> String]
 
+clojure.template/apply-template [ (t/HVec [t/Any *]) (t/HVec [t/Any *])(t/List t/Any ) -> (t/HVec [t/Any *])]
```

Comment:
Why did you use `t/HVec` and not just `t/Vec`? I'm getting used to core.typed and not very familiar with the difference.

Also second argument to `apply-template` can be anything, not just vector:
```clojure
(template/apply-template '[a b] 'a '(1 2)) => 1
(template/apply-template '[a b] "Hello world" '(1 2)) => "Hello world"
(template/apply-template '[a b] '{a b} '(1 2)) => {1 2}
```